kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/DbContentService.java

```diff
@@ -2307,6 +2308,25 @@ public InputStream streamResourceBody(ContentResource resource) throws ServerOve
                 out();
             }
         }
+        
+        /**
+         * Return a URI containing a direct link to the asset.
+         * 
+         * @param rootFolder
+         * @param resource
+         * @return URI representing a direct link to the asset or null
+         */
+        public URI getDirectLink(ContentResource resource)
```

Comment:
There is a case to be made, but as it stands, the FileSystemHandler is only at the DbContentService. I don't think we want to move it to the interface, so I see this as internal to this class.